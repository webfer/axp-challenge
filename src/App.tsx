import { Provider } from 'react-redux';
import Home from './components/Home';
import { store } from './store/store';
import scriptLoader from 'react-async-script-loader';

import { AppProps } from './interfaces/interface';

const { REACT_APP_GOOGLE_MAP_API } = process.env;
const App = ({ isScriptLoaded, isScriptLoadSucceed }: AppProps) => {
  if (isScriptLoaded && isScriptLoadSucceed) {
    return (
      <div data-testid="home-page">
        <Provider store={store}>
          <Home />
        </Provider>
      </div>
    );
  } else {
    return <div></div>;
  }
};

export default scriptLoader([
  `https://maps.googleapis.com/maps/api/js?key=${REACT_APP_GOOGLE_MAP_API}&libraries=places`,
])(App);
