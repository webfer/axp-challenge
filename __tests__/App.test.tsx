import { render } from '@testing-library/react';
import App from '../src/App';

describe('Testing in <App/>', () => {
  test('Should match the snapshot', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
