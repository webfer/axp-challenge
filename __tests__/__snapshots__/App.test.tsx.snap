// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <App/> Should match the snapshot 1`] = `
<div>
  <div />
</div>
`;
